src/libPMacc/include/particles/tasks/TaskParticlesReceive.hpp

```diff
@@ -49,15 +57,25 @@ namespace PMacc
         {
             state = Init;
             EventTask serialEvent = __getTransactionEvent();
+            HandleExchanged handleExchanged;
+            HandleNotExchanged handleNotExchanged;
 
             for (int i = 1; i < Exchanges; ++i)
             {
-                if (parBase.getParticlesBuffer().hasReceiveExchange(i))
-                {
+                /* Start new transaction */
+                if(canRunInParallel)
```

Comment:
Please remove `if(canRunInParallel)`, to accept this we need some race condition checks with PIConGPU. Thus this is time intensive we should use the known version here.